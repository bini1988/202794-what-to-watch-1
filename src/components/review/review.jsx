import React from "react";
import PropTypes from "prop-types";
import cn from "classnames";

function Review({className, review = {}}) {
  const {id, text, author, date, rating} = review;

  return (
    <div id={`review-${id}`} className={cn(`review`, className)}>
      <blockquote className="review__quote">
        <p className="review__text">{text}</p>
        <footer className="review__details">
          <cite className="review__author">{author}</cite>
          <time className="review__date" dateTime={date}>
            {(new Date(date)).toDateString()}
          </time>
        </footer>
      </blockquote>
      <div className="review__rating">
        {rating}
      </div>
    </div>
  );
}

Review.propTypes = {
  /** Объект ревью */
  review: PropTypes.shape({
    id: PropTypes.number,
    /** Текст ревью */
    text: PropTypes.string,
    /** Автор */
    author: PropTypes.string,
    /** Дата */
    date: PropTypes.string,
    /** Рейтинг */
    rating: PropTypes.number,
  }),
  /** Дополнительный класс к контейнеру */
  className: PropTypes.string,
};

export default Review;
