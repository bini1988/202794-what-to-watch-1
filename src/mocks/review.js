
export default {
  id: 1,
  author: `Kate Muir`,
  rating: 8.9,
  text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`
};
