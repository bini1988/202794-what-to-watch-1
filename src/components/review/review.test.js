import React from "react";
import renderer from "react-test-renderer";
import Review from "./review";
import review from "../../mocks/review";

it(`Review correctly renders default markup`, () => {
  const tree = renderer.create(
      <Review review={review}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
