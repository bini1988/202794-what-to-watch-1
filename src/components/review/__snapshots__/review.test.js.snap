// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review correctly renders default markup 1`] = `
<div
  className="review"
  id="review-1"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Kate Muir
      </cite>
      <time
        className="review__date"
        dateTime="2019-05-08T14:13:56.569Z"
      >
        Wed May 08 2019
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    8.9
  </div>
</div>
`;
